File: c++/Stack/src/stack.cpp
#include "stack.hpp"

Stack::Stack() : head(nullptr), element_numbers(0){

}
Stack::~Stack() {

}
void Stack::push(const int &data) {
  if(this->head == nullptr) {
    this->head = new Node(data);
  }
  Node *new_item = new Node(data);
  new_item->next = this->head;
  this->head = new_item;
  this->element_numbers++;
}
int Stack::pop() {
    int data = this->head->data;
    Node *trash = this->head;
    this->head = this->head->next;
    delete(trash);
    this->element_numbers--;
    return data;
}
bool Stack::contains(const int &data){
    Node *current = this->head;
    while (current->next != nullptr)
    {
        if(current->data == data) {
            return true;
        }
        current = current->next;
    }
    return false;
}
int Stack::size(){
    return this->element_numbers;
}

File: python/Data Structures/Stack/stack.py
class Stack:
    class Node:
        def __init__(self, item):
            self.item = item
            self.next = None

    def __init__(self):
        self.head = None
        self.size = 0

    def put(self,item):
        if(self.head is None):
            self.head = self.Node(item)
        else:
            node = self.Node(item)
            node.next = self.head
            self.head = node
        self.size += 1

    def pop(self):
        if(self.head is not None):
            item = self.head.item
            self.head = self.head.next
            self.size -= 1
            return item

    def show(self):
        current = self.head
        print(current.item)
        while(current.next is not None):
            print(current.next.item)
            current = current.next


if __name__ == '__main__':
    stack = Stack()
    stack.put(2)
    stack.put(4)
    stack.put(6)
    stack.put(8)
    stack.put(10)

    print(stack.pop())
    print(stack.pop())
    print("##############")

    stack.show()
        